### DISCUSSION TITLE
Compatibility of Apollo with Locally Installed Carla 0.9.13

### DISCUSSION THREAD

**Discussion by i-kj** [guardstrikelab/carla_apollo_bridge#147] [Q&A] [answered]
Dear Developers,

I am working on a project that involves running Apollo alongside Carla for autonomous vehicle simulation. I have Carla version 0.9.13 already installed on my local machine, and I am interested in integrating it with Apollo using the Carla-Apollo bridge.

Before proceeding, I would like to confirm a few things regarding the compatibility between Carla 0.9.13 and Apollo,

    Is Carla 0.9.13 fully compatible with the latest version of Apollo, and are there any known issues or limitations that I should be aware of?

    If Carla 0.9.13 is not directly compatible, are there any particular branches or forks of Apollo that are known to work well with this version?

    Regarding the guardstrike/apollo_carla_bridge branch specifically, is it compatible with Carla 0.9.13 or is it tailored for more recent versions of Carla?

    Are there any workarounds, patches, or specific configurations that you would recommend for running Carla 0.9.13 with Apollo?

Here are some details about my setup for context:

    Operating System: Ubuntu 20.04.6 LTS
    Graphics Card: NVIDIA GeForce GTX 1080
    Docker Containers: I have the capability to run both Carla and Apollo in Docker containers and have nvidia-docker2 installed for GPU support.

Your insights would be invaluable for my project, and I appreciate any guidance you can provide.

Thanks in advance.

**Comment by XiaoFei9704** [comment, marked as answer]
It is not completely compatible. For example, the weather elements in Carla cannot be matched in Apollo. The vehicle dynamics parameters of Carla are relatively simple, while the vehicle dynamics parameters of Apollo are very complete.
However, all the elements required for joint simulation can be matched. You can use **carla0.9.13 and apollo8.0/7.0 for full closed-loop** simulation.
